system/Database/BaseBuilder.php

```diff
@@ -671,7 +671,7 @@ public function join(string $table, string $cond, string $type = '', bool $escap
                 $operator = $this->getOperator($condition);
 
                 $cond .= $joints[$i];
-                $cond .= preg_match("/(\(*)?([\[\]\w\.'-]+)" . preg_quote($operator, '/') . '(.*)/i', $condition, $match) ? $match[1] . $this->db->protectIdentifiers($match[2]) . $operator . $this->db->protectIdentifiers($match[3]) : $condition;
+                $cond .= preg_match('/(\(*)?([\[\]\w\.\'-]+)' . preg_quote($operator, '/') . '(.*)/i', $condition, $match) ? $match[1] . $this->db->protectIdentifiers($match[2]) . $operator . $this->db->protectIdentifiers($match[3]) : $condition;
```

Comment:
I'm not sure if this is a separate rule but IMO this was better off with the double quotes and no escape.

Reply:
No yours looks much better 😅